modules/images/image-loading-optimization/detection/detect.js

```diff
@@ -111,7 +113,21 @@ function getElementIndex( element ) {
 	if ( ! element.parentElement ) {
 		return 0;
 	}
-	return [ ...element.parentElement.children ].indexOf( element );
+	const children = [ ...element.parentElement.children ];
+	let index = children.indexOf( element );
+	if ( children.includes( document.getElementById( adminBarId ) ) ) {
+		// TODO: Should detection just be turned off when is_user_logged_in()?
+		--index;
+	}
+	if (
+		children.includes(
+			document.querySelector( '.skip-link.screen-reader-text' )
```

Comment:
This should be cached.

Reply:
Once you switch to generate breadcrumbs on the server,  maybe a shorter index or hash passed to the front end?

Reply:
Yes, exactly. Currently I'm thinking that when detection is needed that the server would add breadcrumbs via a data attribute, for example:

```html
<img data-ilo-breadcrumbs="html,0 body,1 main,2 figure,10 img,0" ...>
```

When detection is not needed, such data attributes would not be present.

Reply:
There is already one:

https://github.com/WordPress/performance/blob/0cca4dd2bf49afcfa9af51feaf58a83f7a5f57c8/modules/images/image-loading-optimization/detection/detect.js#L136

Reply:
Done in the next PR: https://github.com/WordPress/performance/pull/892